api/src/main/java/org/apache/iceberg/Metrics.java

```diff
@@ -120,4 +124,83 @@ public Long recordCount() {
   public Map<Integer, ByteBuffer> upperBounds() {
     return upperBounds;
   }
+
+  /**
+   * Implemented the method to enable serialization of ByteBuffers.
+   * @param out The stream where to write
+   * @throws IOException On serialization error
+   */
+  private void writeObject(ObjectOutputStream out) throws IOException {
+    out.writeObject(rowCount);
+    out.writeObject(columnSizes);
+    out.writeObject(valueCounts);
+    out.writeObject(nullValueCounts);
+
+    writeByteBufferMap(out, lowerBounds);
+    writeByteBufferMap(out, upperBounds);
+  }
+
+  private static void writeByteBufferMap(ObjectOutputStream out, Map<Integer, ByteBuffer> byteBufferMap)
+      throws IOException {
+    if (byteBufferMap == null) {
+      out.writeInt(-1);
+
+    } else {
+      // Write the size
+      out.writeInt(byteBufferMap.size());
+
+      for (Map.Entry<Integer, ByteBuffer> entry : byteBufferMap.entrySet()) {
+        // Write the key
+        out.writeObject(entry.getKey());
+
+        // Write the value
+        if (entry.getValue() != null) {
+          out.writeObject(entry.getValue().array());
```

Comment:
I think this is one of those possibly dangerous things if the underlying implementation of the byte buffer isn't backed by an array. I believe the safe way to handle this is
```
byte[] b = new byte[bb.remaining()];
bb.get(b);
```

Reply:
I do not know how this ByteBuffer is used. I was considering the proposed solution, but I am not entirely sure that we only want to store the remaining part of the ByteBuffer, so I would like to know the opinions of people more familiar with the Iceberg usages of this stuff.
Maybe a check for specific implementation and error message for other options would be better. What do you think?

Reply:
Maybe just checking the hasArray and checking that the offset is 0 would be enough. And throw an exception in other cases. Maybe we can avoid the costly buffer allocation. Or use them only when they are needed.
Good that you brought this up. Started to fix this warning, but then got sidetracked 😄

Reply:
@RussellSpitzer: Fixed with your suggested changes. I have chickened out from using different serialization method for different buffers because I think the added complexity might not worth the gained performance.

Could you please recheck the change?

Thanks,
Peter

Reply:
lgtm!